src/libsyntax/ast.rs

```diff
@@ -479,7 +451,7 @@ pub struct WhereEqPredicate {
 
 /// The set of MetaItems that define the compilation environment of the crate,
 /// used to drive conditional compilation
-pub type CrateConfig = Vec<P<MetaItem>>;
+pub type CrateConfig = HashSet<(Name, Option<Symbol>)>;
```

Comment:
Hmm, we could make `Option<Symbol>` the same size as `Symbol` by using `NonZero`, maybe in a future PR?